arbitrator/include/capabilities_interface.tpp

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2019 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#ifndef __CAPABILITIES_INTERFACE_TPP__
+#define __CAPABILITIES_INTERFACE_TPP__
+
+#include <vector>
+#include <map>
+#include <string>
+#include <functional>
+
+namespace arbitrator 
+{
+    template<typename MSrv>
```

Comment:
Not sure I get the reason why you template this class.

Reply:
It's just this function that's templated, but it's so it could be used for multiple different kinds of services calls (each needing a different kind of MSrv template parameter for their message type).

Reply:
Not in the present, thought in the future I believe it probably will. For example it'll need to use a different service to interact with the cost plugins to get their costs for a given plan.